core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -2622,6 +2623,32 @@ class KafkaApis(val requestChannel: RequestChannel,
         .setTopics(endOffsetsForAllTopics)))
   }
 
+  def handleDynamicConfigs(request: RequestChannel.Request, isIncrementalRequest: Boolean): Unit = {
```

Comment:
This handles the validation, persisting, and forwarding logic for both ZK and KRaft brokers to ensure that configs are validated on the broker that the configs are for before persisting them.